demo/include/demo/change_dot_cluster_behavior.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <arbitration_graphs/behavior.hpp>
+
+#include "environment_model.hpp"
+#include "types.hpp"
+#include "utils/cluster.hpp"
+
+namespace demo {
+
+/**
+ * @brief The ChangeDotClusterBehavior makes pacman move towards the closest dot cluster
+ * that he is not currently inside of.
+ */
+class ChangeDotClusterBehavior : public arbitration_graphs::Behavior<Command> {
+public:
+    using Ptr = std::shared_ptr<ChangeDotClusterBehavior>;
+    using ConstPtr = std::shared_ptr<const ChangeDotClusterBehavior>;
+
+    using Cluster = utils::Cluster;
+    using Clusters = utils::DotClusterFinder::Clusters;
+
+    explicit ChangeDotClusterBehavior(EnvironmentModel::Ptr environmentModel,
+                                      const std::string& name = "ChangeDotClusterBehavior")
+            : Behavior(name), environmentModel_{std::move(environmentModel)} {
+    }
+
+    Command getCommand(const Time& /*time*/) override {
+        std::optional<Path> pathToTargetClusterCenter = environmentModel_->pathTo(targetCluster_->center);
+
+        if (!pathToTargetClusterCenter) {
+            throw std::runtime_error("Failed to compute path to target cluster. Can not provide a sensible command.");
+        }
+
+        return Command{pathToTargetClusterCenter.value()};
+    }
```

Comment:
We might want to move / always do invCond/commCond and getCommand implementations in the cpp file.
It can be confusing why some are in the header, some in the source.

Only exception could be one-liners like `gainControl()` and `loseControl()` in this case here.

Reply:
See 5a6118b9db9c